nimby-vs-yimby/README.md

```diff
@@ -0,0 +1,303 @@
+db_connection = create_engine(db_connection_str)
+
+query = "SELECT * FROM `states`;"
+states_df = pd.read_sql(query, db_connection)
+states_df = states_df.rename(columns={'name': 'state'})
+
+df_counties = pd.merge(df_counties, states_df, on='state')
+df_counties['county_state'] = df_counties['county'] + ', ' + df_counties['code'] 
+del df_counties['county']
+del df_counties['code']
+del df_counties['state']
+
+df_counties.loc[df_counties['fips'] == "51600", "county_state"] = "FAIRFAX CITY, VA"
+df_counties.loc[df_counties['fips'] == "24510", "county_state"] = "BALTIMORE CITY, MD"
+df_counties.to_csv("counties.csv")
+
+db_connection_str = 'mysql+mysqlconnector://rl:trustno1@localhost/us_housing_prices_v2'
+db_connection = create_engine(db_connection_str)
+
+query = """
+SELECT a.*
+FROM `sales` a
+INNER JOIN
+(
+    SELECT   `property_id`, `state`, `property_zip5`, `property_county`, MIN(`sale_datetime`) AS first_sale_datetime
+    FROM     `sales`
+    GROUP BY `property_id`
+) b ON a.property_id = b.property_id AND a.sale_datetime = b.first_sale_datetime
+WHERE b.first_sale_datetime > \"2009-06-30\" AND b.first_sale_datetime < \"2020-01-01\";
+"""
+
+raw_count_df = pd.read_sql(query, db_connection)
+raw_count_df.to_csv("~/raw.csv")
+result_df = pd.DataFrame(raw_count_df)
+result_df = result_df[['state', 'property_county', 'sale_datetime', 'property_id']]
+result_df = result_df.rename(columns={'property_county': 'county', 'state':'code'})
+result_df['county'] = result_df['county'].apply(cleanup_county)
+result_df['county_state'] = result_df['county'] + ', ' + result_df['code'] 
+del result_df['county']
+del result_df['code']
+result_df.to_csv("~/result.csv")
+
+counts_by_county = result_df['county_state'].value_counts()
+
+df_counts_by_county = pd.DataFrame.from_records([counts_by_county.to_dict()]).transpose()
+df_counts_by_county.reset_index(inplace=True)
+df_counts_by_county = df_counts_by_county.rename(columns={'index': 'county_state', 0: 'n'})
+df_counts_by_county = df_counts_by_county.sort_values('n', ascending=False)
+
+df_counts_by_county = pd.merge(df_counts_by_county, df_counties, on='county_state')
+
+df_counts_by_county['per_capita'] = df_counts_by_county['n'] / df_counts_by_county['pop2010']
+df_counts_by_county['per_area'] = df_counts_by_county['n'] / df_counts_by_county['area_2010']
+
+per_capita_mean = float(df_counts_by_county['per_capita'].mean())
+per_capita_stdev = float(df_counts_by_county['per_capita'].std())
+
+df_counts_by_county['per_capita_stdevs_from_mean'] = (df_counts_by_county['per_capita'] - per_capita_mean) / per_capita_stdev
+
+per_area_mean = float(df_counts_by_county['per_area'].mean())
+per_area_stdev = float(df_counts_by_county['per_area'].std())
+
+df_counts_by_county['per_area_stdevs_from_mean'] = (df_counts_by_county['per_area'] - per_area_mean) / per_area_stdev
+df_counts_by_county = df_counts_by_county.sort_values('per_capita', ascending=False)
+
+df_counts_by_county.to_csv("~/counts.csv")
+```
```

Comment:
Not sure if this should be inlined in the blog. Maybe a link to a Google colab?

Reply:
DoltHub blog has a way to fold blocks of code so that they don't get in the way that was used in previous posts (not sure if I'm using the correct syntax for that).

Google Colab is unfortunately out of the question due to computational requirements involved to work 112GB database.